pathio.go

```diff
@@ -55,7 +55,7 @@ func s3FileReader(path string) (io.ReadCloser, error) {
 	}
 
 	// Look up region in S3
-	region, err := getRegionForBucket(bucket)
+	region, err := getRegionForBucket(getS3Api(), bucket)
```

Comment:
Rather than instantiate the `S3Api` on each invocation I think we could just move it up into a struct that we mock out completely.

E.g.
```go
func s3FileReader(awsClient s3Handler, path string)
```
`...`
```go
region, err := getRegionForBucket(awsClient, bucket)
```